benchmark_test.go

```diff
@@ -0,0 +1,62 @@
+package cid_test
+
+import (
+	"github.com/ipfs/go-cid"
+	"github.com/multiformats/go-multihash"
+	"math/rand"
+	"testing"
+)
+
+// BenchmarkIdentityCheck benchmarks checking whether a CIDv1 has multihash.IDENTITY
+// code via two methods: 1) Cid.Prefix() and 2) decoding the Cid.Hash().
+func BenchmarkIdentityCheck(b *testing.B) {
+	rng := rand.New(rand.NewSource(1413))
+	cv1 := generateCidV1(b, rng, multihash.IDENTITY)
+
+	b.SetBytes(int64(cv1.ByteLen()))
+	b.ReportAllocs()
+	b.ResetTimer()
+
+	b.Run("Prefix", func(bb *testing.B) {
+		bb.RunParallel(func(pb *testing.PB) {
```

Comment:
I wouldn't name the inner one `bb`, because then it's too easy to make the mistake of calling methods on the wrong (parent) `b`, just like you did on line 24. If this sub-benchmark fails, only it should fail, not the entire parent benchmark.

Reply:
This is one example where shadowing is a good thing: it prevents using the wrong "level".